### ISSUE TITLE
When key does not exists i get Index was outside the bounds of the array

### ISSUE THREAD

**Issue by zulander1** [redis/redis-om-dotnet#169]
For some reason, in production, when the key does not exists it is throwing an Index was outside the bounds of the array, this happens when we are enumerating though a large collection... 

![image](https://user-images.githubusercontent.com/12099546/182716192-8e54e612-1253-40eb-b7bd-e19eaba91f14.png)

**Comment by slorello89** [comment]
Hmmm, what code is precipitating this? Sounds like a server error the constructor isn't handling appropriately.

**Comment by zulander1** [comment]
@slorello89 

```
        public async Task<IEnumerable<GroupMemberModel>> GetAsync(Expression<Func<GroupMemberModel, bool>> predicate)
        {
            return (await _groupMembers.Where(predicate).ToListAsync()).Select(x => x);
        }
```

**Comment by slorello89** [comment]
I'm pretty sure this is the same as #334 - so it will be fixed by #367.
